src/tree_state.rs

```diff
@@ -365,3 +365,19 @@ where
         }
     }
 }
+
+impl<Identifier> Default for TreeState<Identifier> {
+    fn default() -> Self {
+        Self {
+            offset: 0,
+            opened: HashSet::default(),
+            selected: Vec::default(),
+            ensure_selected_in_view_on_next_render: false,
+
+            last_area: Rect::default(),
+            last_biggest_index: 0,
+            last_identifiers: Vec::default(),
+            last_rendered_identifiers: Vec::default(),
```

Comment:
Sadly trait impl methods (like `default()`) can not be const (yet). When its auto-implemented (`derive(Default)`) it's not worth the effort for me (I still try to prefer (const) new methods). When its manually implemented I think it's a good idea to be a bit more explicit and use const when possible. LLVM will figure it out itself but using const there should be the explicit way of telling LLVM to do that on compile time.

Sadly `HashSet::new` is not const, otherwise I would have prefered a const new method.

```suggestion
            opened: HashSet::new(),
            selected: Vec::new(),
            ensure_selected_in_view_on_next_render: false,

            last_area: Rect::ZERO,
            last_biggest_index: 0,
            last_identifiers: Vec::new(),
            last_rendered_identifiers: Vec::new(),
```

Reply:
Good point, I have changed that.